iree/compiler/Conversion/HLOToLinalg/test/linalg_tensor_to_buffer.mlir

```diff
@@ -163,3 +163,212 @@ module {
+//       CHECK:   linalg.generic {{.*}} %[[T2]], %[[T0]]
+//       CHECK:   %[[T3:.*]] = linalg.reshape %[[T0]]
+//       CHECK:   linalg.copy(%[[T3]], %[[T1]])
+//       CHECK:   return
+
+// -----
+
+#map0 = affine_map<(d0, d1) -> (d0, d1)>
+#map1 = affine_map<(d0, d1, d2) -> (d0, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d2)>
+
+module {
+  func @store_reshape_src_and_result_2() {
+    %c0 = constant 0 : index
+    %0 = hal.interface.load.tensor @legacy_io::@arg0,
+                                   offset = %c0 : tensor<2x4xf32>
+    %1 = linalg.generic
+           {args_in = 1 : i64, args_out = 1 : i64,
+            indexing_maps = [#map0, #map0],
+            iterator_types = ["parallel", "parallel"]} %0 {
+    ^bb0(%arg0: f32):  // no predecessors
+      %2 = tanh %arg0 : f32
+      linalg.yield %2 : f32
+    }: tensor<2x4xf32> -> tensor<2x4xf32>
+    %3 = linalg.tensor_reshape %1 [#map1, #map2] : tensor<2x4xf32> into tensor<1x2x4xf32>
+    %4 = linalg.tensor_reshape %1 [#map1, #map2] : tensor<2x4xf32> into tensor<1x2x4xf32>
+    %5 = linalg.tensor_reshape %1 [#map1, #map2] : tensor<2x4xf32> into tensor<1x2x4xf32>
+    hal.interface.store.tensor %3, @legacy_io::@ret0, offset = %c0 : tensor<1x2x4xf32>
+    hal.interface.store.tensor %4, @legacy_io::@ret1, offset = %c0 : tensor<1x2x4xf32>
+    hal.interface.store.tensor %5, @legacy_io::@ret2, offset = %c0 : tensor<1x2x4xf32>
+    return
+  }
+  hal.interface @legacy_io attributes {sym_visibility = "private"} {
+    hal.interface.binding @arg0, set=0, binding=0,
+                                 type="StorageBuffer", access="Read"
+    hal.interface.binding @ret0, set=0, binding=1,
+                                 type="StorageBuffer", access="Write|Discard"
+    hal.interface.binding @ret1, set=0, binding=2,
+                                 type="StorageBuffer", access="Write|Discard"
+    hal.interface.binding @ret2, set=0, binding=3,
+                                 type="StorageBuffer", access="Write|Discard"
+  }
+}
+
+// CHECK-LABEL: func @store_reshape_src_and_result_2
+//       CHECK:   %[[T0:.*]] = iree.placeholder {{.*}} @legacy_io::@ret0
+//       CHECK:   %[[T1:.*]] = linalg.reshape %[[T0]]
+//       CHECK:   %[[T2:.*]] = iree.placeholder {{.*}} @legacy_io::@ret1
+//       CHECK:   %[[T3:.*]] = iree.placeholder {{.*}} @legacy_io::@ret2
+//       CHECK:   %[[T4:.*]] = iree.placeholder {{.*}} @legacy_io::@arg0
+//       CHECK:   linalg.generic {{.*}} %[[T4]], %[[T1]]
+//       CHECK:   linalg.copy(%[[T0]], %[[T2]])
+//       CHECK:   linalg.copy(%[[T0]], %[[T3]])
+//       CHECK:   return
+
+// -----
+
+#map0 = affine_map<(d0, d1, d2, d3) -> (d0, d1)>
+#map1 = affine_map<(d0, d1, d2, d3) -> (d2, d3)>
+#map2 = affine_map<(d0, d1) -> (d0, d1)>
+
+module attributes {spv.target_env = #spv.target_env<#spv.vce<v1.3, [Shader], [SPV_KHR_storage_buffer_storage_class]>, {max_compute_workgroup_invocations = 128 : i32, max_compute_workgroup_size = dense<[128, 128, 64]> : vector<3xi32>}>} {
```

Comment:
nit: remove the attributes for module. This is not needed for this pass.

Reply:
Done